### PR TITLE
Pressure Tanks no longer runtime when shuttleRotated, Remedies constant CI fails

### LINKED ISSUE #71092: Omni pressure tanks runtime at shuttle rotation
```
[2022-11-05 09:28:41.475] runtime error: list index out of bounds
 - proc name: shuttleRotate (/obj/machinery/atmospherics/shuttleRotate)
 -   source file: shuttle_rotate.dm,94
 -   usr: null
 -   src: the omni pressure tank (Air) (/obj/machinery/atmospherics/components/tank/air)
 -   src.loc: the plating (167,169,8) (/turf/open/floor/plating)
 -   call stack:
 - the omni pressure tank (Air) (/obj/machinery/atmospherics/components/tank/air): shuttleRotate(270, null)
 - the omni pressure tank (Air) (/obj/machinery/atmospherics/components/tank/air): afterShuttleMove(hyperspace (105,28,12) (/turf/open/space/transit/west), /list (/list), 2, 8, 8, 270)
 - the omni pressure tank (Air) (/obj/machinery/atmospherics/components/tank/air): afterShuttleMove(hyperspace (105,28,12) (/turf/open/space/transit/west), /list (/list), 2, 8, 8, 270)
 - the omni pressure tank (Air) (/obj/machinery/atmospherics/components/tank/air): afterShuttleMove(hyperspace (105,28,12) (/turf/open/space/transit/west), /list (/list), 2, 8, 8, 270)
 - White Ship (/obj/docking_port/mobile): cleanup runway(the custom location (/obj/docking_port/stationary), /list (/list), /list (/list), /list (/list), /list (/list), 270, 8, Hyperspace (/area/shuttle/transit))
 - White Ship (/obj/docking_port/mobile): initiate docking(the custom location (/obj/docking_port/stationary), 8, 0)
 - White Ship (/obj/docking_port/mobile): check()
 - Shuttle (/datum/controller/subsystem/shuttle): fire(0)
 - Shuttle (/datum/controller/subsystem/shuttle): ignite(0)
```

## Reproduction:

1. Find a pilotable shuttle
2. If it doesnt have a pressure tank already, add one
3. Rotate your ship via custom location/docking/whatever
4. Runtime

### PR DESCRIPTION
**Pull request by tontyGH**
## About The Pull Request

Fixes #71092 

## Why It's Good For The Game

Flaky tests bad, fixman good...

## Changelog

:cl:
fix: Pubby's whiteship no longer breaks when it tries to dock
/:cl:

### CODE REVIEW

File: code/__HELPERS/type2type.dm [hunk 1]
```diff
@@ -99,6 +99,11 @@ GLOBAL_LIST_INIT(modulo_angle_to_dir, list(NORTH,NORTHEAST,EAST,SOUTHEAST,SOUTH,
 		else
 			return null
 
+///Returns a single dir rotated by x degrees.
+#define rotate_dir(dir, rotation) ( angle2dir( dir2angle(dir) + rotation ) )
+///Returns a single dir rotated by x degrees, adhering to the cardinal directions.
+#define rotate_dir_cardinal(dir, rotation) ( angle2dir_cardinal ( dir2angle(dir) + rotation ) )
```
Comment (LemonInTheDark) [inline code/__HELPERS/type2type.dm:105]: use turn() no? at least for rotate dir. also replace the decal code that does rotate_dir_cardinal with this macro (or one renamed to fit the turn() pattern)
also macros should typically be written with upper case

File: code/modules/atmospherics/machinery/components/tank.dm [hunk 2]
```diff
@@ -151,28 +154,53 @@
 	refresh_window()
 
 ///////////////////////////////////////////////////////////////////
-// Pipenet stuff
+// Port stuff
+
+/**
+ * Enables/Disables a port direction in [./var/open_ports]. \
+ * Use this, then call [./proc/set_init_directions()] instead of setting initialize_directions directly \
+ * This system exists because having some initialize_directions set to null breaks rotation
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/components/tank.dm:162]: \ here is redundant

File: code/modules/atmospherics/machinery/components/tank.dm [hunk 3]
```diff
@@ -151,28 +154,53 @@
 	refresh_window()
 
 ///////////////////////////////////////////////////////////////////
-// Pipenet stuff
+// Port stuff
+
+/**
+ * Enables/Disables a port direction in [./var/open_ports]. \
+ * Use this, then call [./proc/set_init_directions()] instead of setting initialize_directions directly \
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/components/tank.dm:161]: I don't think? this syntax works, does it?

File: code/modules/atmospherics/machinery/components/tank.dm [hunk 4]
```diff
@@ -151,28 +154,53 @@
...
+ * This system exists because having some initialize_directions set to null breaks rotation
+ */
+/obj/machinery/atmospherics/components/tank/proc/set_portdir_relative(relative_port_dir, enable)
+
+	if(isnull(enable)) //toggle it if we weren't given a state to set it to
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/components/tank.dm:166]: add a new proc that does this, do not make toggling implicit behavior off null it mixes metaphor

File: code/modules/atmospherics/machinery/components/tank.dm [hunk 5]
```diff
@@ -151,28 +154,53 @@
...
+		enable = ((initialize_directions & relative_port_dir) ? FALSE : TRUE)
+
+	// Rotate the given dir so that it's relative to north
+	var/port_dir
+	if(dir == NORTH)
+		port_dir = relative_port_dir
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/components/tank.dm:172]: is this required?

File: code/modules/atmospherics/machinery/components/tank.dm (hunk 2 above)
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/components/tank.dm:162]: is this going to be useful to people into the future?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
What is it that you don't like about this?

**Review by Ghommie — APPROVED** [review]

**Pull request merged**